.ci/workspace-startup/start-workspace-reporter/src/main/java/com/redhat/che/start_workspace_reporter/ReporterMain.java

```diff
@@ -0,0 +1,758 @@
+        Map<String, Float> changes = new HashMap<>();
+        changes.put("pvc_start_avg", getPercentageDifference(oldPvcStartAvg, newPvcStartAvg));
+        changes.put("pvc_start_max", getPercentageDifference(oldPvcStartMax, newPvcStartMax));
+        changes.put("eph_start_avg", getPercentageDifference(oldEphStartAvg, newEphStartAvg));
+        changes.put("eph_start_max", getPercentageDifference(oldEphStartMax, newEphStartMax));
+        changes.put("pvc_stop_avg", getPercentageDifference(oldPvcStopAvg, newPvcStopAvg));
+        changes.put("pvc_stop_max", getPercentageDifference(oldPvcStopMax, newPvcStopMax));
+        changes.put("eph_stop_avg", getPercentageDifference(oldEphStopAvg, newEphStopAvg));
+        changes.put("eph_stop_max", getPercentageDifference(oldEphStopMax, newEphStopMax));
+        return changes;
+    }
+
+    private static void storeZabbixValueForComparison(AtomicReference<Float> storeValue, AtomicReference<Float> currentValue) {
+        storeValue.set(currentValue.get());
+        currentValue.set(MINMAX_INIT_VALUE);
+    }
+
+    private static float getPercentageDifference(AtomicReference<Float> oldValue, AtomicReference<Float> newValue) {
+        return (newValue.get() - oldValue.get()) / oldValue.get() * 100;
+    }
+
+    private static boolean grabHistoryDataFromZabbix(HttpRequestWrapper wrapper, HttpRequestWrapperResponse response, JSONRPCRequest getHistoryRequest, List<ZabbixHistoryMetricsEntry> zabbixHistoryResults) {
+        try {
+            HttpResponse tmp = wrapper.post("/api_jsonrpc.php", ContentType.APPLICATION_JSON.toString(), getHistoryRequest.toString());
+            response = new HttpRequestWrapperResponse(tmp);
+        } catch (IOException e) {
+            LOG.log(Level.SEVERE, "Failed to contact zabbix on devshift.net:" + e.getLocalizedMessage());
+        } catch (IllegalArgumentException e) {
+            LOG.log(Level.SEVERE, "Wrapper failed to parse HtppResponse:" + e.getLocalizedMessage());
+        }
+        if (response != null) {
+            if (responseSuccess(response)) {
+                LOG.log(Level.INFO, "Zabbix getHistory request successful.");
+                try {
+                    JsonArray tmp = gson.fromJson(response.asJSONRPCResponse().getResult(), JsonArray.class);
+                    tmp.forEach(e -> zabbixHistoryResults.add(gson.fromJson(e, ZabbixHistoryMetricsEntry.class)));
+                } catch (IOException e) {
+                    LOG.log(Level.SEVERE, "Failed to parse response into value list:"+e.getLocalizedMessage());
+                    return false;
+                }
+            } else {
+                LOG.log(Level.SEVERE, "Zabbix getHistory request failed.");
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private static void calculateZabbixResults(List<ZabbixHistoryMetricsEntry> zabbixHistoryResults,
```

Comment:
Methods with so many arguments are considered a bad practice. Please try to keep your code more readable and easy to maintain. This is also a technical debt being created here. The code (overall) is also having a lot of duplicities.

Reply:
we decided to remove the reporter from this PR for now, will provide a new PR once the code is refactored.